### PR TITLE
Task 2 compare big xmls

### PR DESCRIPTION
**Pull request by akulebyakin**
Task 2 completed

### CODE REVIEW

File: xml-compare/src/test/java/tests/TaskTwoRunTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if (zipDataFile == null) {
+            log.info("The property 'zip_data_file' is not set, so data file unzipping was skipped");
+            return;
+        }
+
+        File zipFile = new File(zipDataFile);
+        String goldData = AppProperties.getProperty("task_two_gold_data_folder");
+        String outputDataFolder = AppProperties.getProperty("task_two_output_data_folder");
+        String goldDataRegex = "A\\d?+.xml";
+        String outputDataRegex = "B\\d?+.xml";
```
Comment (pqtech) [inline xml-compare/src/test/java/tests/TaskTwoRunTest.java]: It seems for me that these patterns won't be working for files like A10.xml, A11.xml etc or for A01.xml, A02.xml

Comment (akulebyakin) [reply xml-compare/src/test/java/tests/TaskTwoRunTest.java]: You are 100% right. Fixed, thank you! :)

File: xml-compare/src/test/java/testdata/TestData.java [hunk 2]
```diff
@@ -4,19 +4,81 @@
...
         return new Object[][]{
                 {goldData, outputData, ignoreNodes}
         };
     }
 
+    @DataProvider(name = "getXmlFilesFromGoldDataAndOutputDataFolders", parallel = true)
+    public static Object[][] getXmlFilesFromGoldDataAndOutputDataFolders() {
+        String gold_data_folder = AppProperties.getProperty("task_two_gold_data_folder");
+        String output_data_folder = AppProperties.getProperty("task_two_output_data_folder");
+        String[] ignoreNodesDefinitions = AppProperties.getProperty("task_two_ignore_nodes_definitions", "").split("([,;])");
```
Comment (pqtech) [inline xml-compare/src/test/java/testdata/TestData.java:33]: I think it is better to keep lines not longer then 100 chars. Java Code Conventions says about 80 chars. I prefer to keep it not longer than 100.

File: xml-compare/src/test/java/testdata/TestData.java [hunk 3]
```diff
@@ -4,19 +4,81 @@
...
 
         return new Object[][]{
                 {goldData, outputData, ignoreNodes}
         };
     }
 
+    @DataProvider(name = "getXmlFilesFromGoldDataAndOutputDataFolders", parallel = true)
+    public static Object[][] getXmlFilesFromGoldDataAndOutputDataFolders() {
+        String gold_data_folder = AppProperties.getProperty("task_two_gold_data_folder");
+        String output_data_folder = AppProperties.getProperty("task_two_output_data_folder");
```
Comment (pqtech) [inline xml-compare/src/test/java/testdata/TestData.java]: These two variables should have names like `goldDataFolder` and `outputDataFolder` :)

File: xml-compare/src/main/java/utils/ZipUtils.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.zip.ZipFile;
+
+@Log4j2
+public class ZipUtils {
+
+    public static void extractGoldDataAndOutputData(@NonNull File zipFile, @NonNull String goldDataFolder,
+                                                    @NonNull String outputDataFolder, @NonNull String goldDataRegex,
+                                                    @NonNull String outputDataRegex) {
+
+        if (!zipFile.exists()) log.error("File '{}' does not exist", zipFile);
```
Comment (pqtech) [inline xml-compare/src/main/java/utils/ZipUtils.java]: Java Code Convention prefers use braces for if statements even in this short line for better code :)
I also suggest you to try to use maven-checkstyle-plugin. It really helps to find some code style violations and fix it easily. For your code it is about 40 violations (using shoggy's checkstyle configuration).

File: xml-compare/src/test/java/testdata/TestData.java (hunk 3 above)
Comment (akulebyakin) [reply xml-compare/src/test/java/testdata/TestData.java]: Done :)

File: xml-compare/src/test/java/testdata/TestData.java (hunk 2 above)
Comment (akulebyakin) [reply xml-compare/src/test/java/testdata/TestData.java:33]: Yeah I agree. Changed line lengths where I found it possible and appropriate.

File: xml-compare/src/main/java/utils/ZipUtils.java (hunk 4 above)
Comment (akulebyakin) [reply xml-compare/src/main/java/utils/ZipUtils.java]: Fixed)
Well, don't really like all of the rules in there, but I fixed most of the plugin comments

**Pull request merged**